src/python/pants/backend/python/packaging/pyoxidizer/rules_integration_test.py

```diff
@@ -71,6 +71,46 @@ def test_end_to_end() -> None:
         assert bin_stdout == b"Hello world!\n"
 
 
+def test_run_binary() -> None:
```

Comment:
Should this be merged with the `test_end_to_end` to save the long compilation time for PyOx?

Reply:
Yeah for sure I think this should be in `test_end_to_end`. I recommend still keeping the `package` + `subprocess.run()` though, we want to be confident both `package` and `run` are working.